File: Problems/0453_Minimum_Moves_to_Equal_Array_Elements/Minimum_Moves_to_Equal_Array_Elements.cs
using System;

public class Solution
{
    /*
    public int MinMoves_submission(int[] nums)
    {
        long sum = 0;

        foreach (var num in nums)
            sum += num;

        sum -= nums.Min() * nums.Length;

        return (int) sum;
    }
    */
    public int MinMoves(int[] nums)
    {
        return sum(nums) - min(nums) * nums.Length;
    }

    public int sum(int[] nums)
    {
        int sum = 0;
        for (int i = 0; i < nums.Length; ++i)
        {
            sum += nums[i];
        }

        return sum;
    }

    public int min(int[] nums)
    {
        int min = int.MaxValue;
        for (int i = 0; i < nums.Length; ++i)
        {
            if (nums[i] < min)
                min = nums[i];
        }

        return min;
    }
    public int[] str_to_int_array(string[] flds)
    {
        if (flds.Length <= 0)
            return null;

        int[] nums = new int[flds.Length];

        for (int i = 0; i < flds.Length; ++i)
        {
            nums[i] = int.Parse(flds[i]);
        }

        return nums;
    }

    public string output_points(int[] nums)
    {
        if (nums.Length <= 0)
            return "";

        string resultStr = "[" + nums[0].ToString();

        for (int i = 1; i < nums.Length; ++i)
        {
            resultStr += "," + nums[i].ToString();
        }

        return resultStr + "]";
    }

    public void Main(string args)
    {
        string temp = args.Replace("[","").Replace("]","").Trim();
        string[] flds = temp.Split(',');
        int[] nums = str_to_int_array(flds);

        Console.WriteLine("nums = " + output_points(nums));
        
        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        int result = MinMoves(nums);
        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms");
    }
}

File: Problems/0453_Minimum_Moves_to_Equal_Array_Elements/Project_Python3/Minimum_Moves_to_Equal_Array_Elements.py
# coding: utf-8

import os
import sys
import time
#from collections import defaultdict
import collections

class Solution:
    def minMoves(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sum(nums) - min(nums) * len(nums)
    
    def get_Max(self, nums):
        max = -1
        for i in range(0, len(nums)):
            if nums[i] > max:
                max = nums[i]
        return max


def str_to_int_array(flds):
    nums = [0]*len(flds)

    if len(flds) <= 0:
        return None
    for i in range(0, len(flds)):
        nums[i] = int(flds[i])

    return nums

def print_array(nums):
    resultStr = "[" + str(nums[0])
    for i in range(1, len(nums)):
        resultStr += "," + str(nums[i])
    return resultStr + "]"

def main():
    argv = sys.argv
    argc = len(argv)

    if (argc < 2):
        print("Usage: python %s <testdata.txt>" %(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("%s not found..." %argv[1])
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        print("argv[1] = %s" %temp)
        loop_main(temp)
    #    print("Hit Return to continue...")
    #    input()

def loop_main(temp):
    var_str = temp.replace("[","").replace("]","").rstrip()
    flds = var_str.split(",")
    nums = str_to_int_array(flds)
    print("%s" %(print_array(nums)))

    time0 = time.time()

    sl = Solution()
    result = sl.minMoves(nums)

    print("result = %d" %result)

    time1 = time.time()
    print("Execute time ... : %f[s]" %(time1 - time0))
    print()

if __name__ == "__main__":
    main()
